PostUploadHandler sometimes crashes during auto-upload of a draft

Running the following steps sometimes end up with a crash for me:

1. Turn on airplane mode
2. Go to post list - drafts
3. Add a new draft with some title and/or content
4. Tap back which should say "Local draft"
5. Turn off airplane mode and immediately refresh the post list (not sure if this part is related or not)
6. Profit (crash)

Here is my test: https://cloudup.com/cRTp-q_JQfA

I was able to reproduce the crash a couple times, but it doesn't happen every time for me.

Edit: This test was run with a WordPress.com account.

```
--------- beginning of crash
2019-09-25 15:16:50.800 14479-14479/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.wordpress.android, PID: 14479
    org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
        at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:527)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:509)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:501)
        at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int org.wordpress.android.fluxc.model.PostModel.getLocalSiteId()' on a null object reference
        at org.wordpress.android.ui.uploads.PostUploadHandler.onPostChanged(PostUploadHandler.java:644)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:507)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:501) 
        at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
2019-09-25 15:16:50.800 14479-14845/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: pool-3-thread-20
    Process: org.wordpress.android, PID: 14479
    org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
        at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:527)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:509)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:501)
        at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
        at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:527)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:509)
        at org.greenrobot.eventbus.EventBus.postToSubscription(EventBus.java:451)
        at org.greenrobot.eventbus.EventBus.postSingleEventForEventType(EventBus.java:411)
        at org.greenrobot.eventbus.EventBus.postSingleEvent(EventBus.java:384)
        at org.greenrobot.eventbus.EventBus.post(EventBus.java:265)
        at org.wordpress.android.fluxc.Dispatcher.emitChange(Dispatcher.java:41)
        at org.wordpress.android.fluxc.store.Store.emitChange(Store.java:32)
        at org.wordpress.android.fluxc.store.PostStore.handleRemoteAutoSavedPost(PostStore.java:1021)
        at org.wordpress.android.fluxc.store.PostStore.onAction(PostStore.java:632)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:507)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:501) 
        at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
     Caused by: java.lang.IllegalStateException: post must not be null
        at org.wordpress.android.ui.posts.PostListEventListener.onPostChanged(PostListEventListener.kt:142)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:507)
        at org.greenrobot.eventbus.EventBus.postToSubscription(EventBus.java:451) 
        at org.greenrobot.eventbus.EventBus.postSingleEventForEventType(EventBus.java:411) 
        at org.greenrobot.eventbus.EventBus.postSingleEvent(EventBus.java:384) 
        at org.greenrobot.eventbus.EventBus.post(EventBus.java:265) 
        at org.wordpress.android.fluxc.Dispatcher.emitChange(Dispatcher.java:41) 
        at org.wordpress.android.fluxc.store.Store.emitChange(Store.java:32) 
        at org.wordpress.android.fluxc.store.PostStore.handleRemoteAutoSavedPost(PostStore.java:1021) 
        at org.wordpress.android.fluxc.store.PostStore.onAction(PostStore.java:632) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:507) 
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:501) 
        at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764)
```

--- a/WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadStarter.kt
+++ b/WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadStarter.kt
@@ -155,12 +155,15 @@ class UploadStarter @Inject constructor(
                             false
                         } else {
                             trackAutoUploadAction(action, it.status)
+                            AppLog.d(
+                                    AppLog.T.POSTS,
+                                    "UploadStarter for post title: ${it.title}, action: $action"
+                            )
                             true
                         }
                     }
                     .toList()
                     .forEach { post ->
-                        AppLog.d(AppLog.T.POSTS, "UploadStarter for post title: ${post.title}")
                         dispatcher.dispatch(UploadActionBuilder.newIncrementNumberOfAutoUploadAttemptsAction(post))
                         uploadServiceFacade.uploadPost(
                                 context = context,

--- a/WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadStarter.kt
+++ b/WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadStarter.kt
@@ -145,26 +145,27 @@ class UploadStarter @Inject constructor(
     private suspend fun upload(site: SiteModel) = coroutineScope {
         try {
             mutex.lock()
-            val posts = postStore.getPostsWithLocalChanges(site)
-
-            posts
+            postStore.getPostsWithLocalChanges(site)
                     .asSequence()
-                    .filter {
-                        val action = uploadActionUseCase.getAutoUploadAction(it, site)
-                        if (action == DO_NOTHING) {
-                            false
-                        } else {
-                            trackAutoUploadAction(action, it.status)
-                            AppLog.d(
-                                    AppLog.T.POSTS,
-                                    "UploadStarter for post title: ${it.title}, action: $action"
-                            )
-                            true
-                        }
+                    .map { post ->
+                        val action = uploadActionUseCase.getAutoUploadAction(post, site)
+                        Pair(post, action)
+                    }
+                    .filter { (_, action) ->
+                        action != DO_NOTHING
                     }
                     .toList()
-                    .forEach { post ->
-                        dispatcher.dispatch(UploadActionBuilder.newIncrementNumberOfAutoUploadAttemptsAction(post))
+                    .forEach { (post, action) ->
+                        trackAutoUploadAction(action, post.status)
+                        AppLog.d(
+                                AppLog.T.POSTS,
+                                "UploadStarter for post title: ${post.title}, action: $action"
+                        )
+                        dispatcher.dispatch(
+                                UploadActionBuilder.newIncrementNumberOfAutoUploadAttemptsAction(
+                                        post
+                                )
+                        )
                         uploadServiceFacade.uploadPost(
                                 context = context,
                                 post = post,

--- a/build.gradle
+++ b/build.gradle
@@ -106,5 +106,5 @@ buildScan {
 
 ext {
     daggerVersion = '2.22.1'
-    fluxCVersion = '6875566c14de5c2d12be6ddc6d68d04bb3338e06'
+    fluxCVersion = 'fcd13311de5274cbadb13a9bd611ae65b71c4712'
 }

